apps/yggdrasil/priv/repo/migrations/20160229163000_create_role_resource.exs

```diff
@@ -0,0 +1,17 @@
+defmodule Yggdrasil.Repo.Migrations.CreateRoleResource do
+  use Ecto.Migration
+
+  def change do
+    create table(:roles_resources) do
+      add :role_id, references(:roles, on_delete: :nothing)
+      add :resource_id, references(:resources, on_delete: :nothing)
+      add :permission_id, references(:permissions, on_delete: :nothing)
+
+      timestamps
+    end
+    create index(:roles_resources, [:role_id])
+    create index(:roles_resources, [:resource_id])
+    create index(:roles_resources, [:permission_id])
```

Comment:
So I don't think we can do composite keys just yet through ecto.

this sounds like it is https://github.com/elixir-lang/ecto/pull/1210and I thought this was in 1.0 or one of them, but apparently it's not.

I do see it mentioned in the [changelog](https://github.com/elixir-lang/ecto/blob/master/CHANGELOG.md).

I should be able to do this when it lands I think. According to the PR above.

create(:roles_resource, primary_key: false) do
  add :role_id, references(:roles), primary_key: true
  add :resource_id references(:resources), primary_key: true
  add :permission_id references(:permissions), primary_key: true
end

but this does not work, the adapter chokes. So for now I can eliminate the primary key, and just leave it as is until ecto 2.0